les/odr_requests.go

```diff
@@ -125,6 +128,46 @@ func (r *BlockRequest) Validate(db ethdb.Database, msg *Msg) error {
 	return nil
 }
 
+//TODO(lucas): Comments
+type HeaderRequest light.HeaderRequest
+
+func (r *HeaderRequest) GetCost(peer *peer) uint64 {
+	return peer.GetRequestCost(GetBlockHeadersMsg, 1)
+}
+
+func (r *HeaderRequest) CanSend(peer *peer) bool {
+	// TODO(lucas): `HasBlock` doesn't work for the hash only case
```

Comment:
Out of curiosity, since the `HeaderRequest` is is serving `odr_utils.GetHeaderByHash`, how does the request figure out if the peer can serve it?

Reply:
Yeah this was something I wasn't sure how to implement - this change is in `les/fetcher.peerHasBlock` to optimistically return true if the provided blocknumber is 0, otherwise it gets confused with the genesis block (`number` is uint64 so I can't provide a clearer sentinel value here).
This will lead to some false positives, but that seemed ok to me. Would appreciate your input

Reply:
Ah ok, the change if `fetcher.peerHasBlock` makes much more sense to me now.

I can't really think of any good ideas other than the obvious (e.g. change the `fetcher.peerHasBlock`'s `number` parameter to have type `*int64` and then check if it's nil).  That may be worth doing if there are not that many `fetcher.peerHasBlock` invocations that you need to change.

Reply:
Implemented this now, please take a look. Makes a lot of things a bit uglier with `&`s and `*`s but works for my tests.